### DISCUSSION TITLE
Markblocks tool cannot find blocks when using GCS storage

### DISCUSSION THREAD

**Discussion by ronanbarrett** [grafana/mimir#3679] [Help and support]
#### Describe the bug
The `markblocks` tool bundled with mimir doesn't find blocks that exist in GCS storage.

#### To Reproduce
1. Mimir writes blocks to storage and we can retrieve them e.g. `gsutil ls -l gs://my-gcp-project/blocks/my-tenant/01GHF3BV8D6JNT8X88Q07F9FZK`
2. Run `markblocks` to delete a block e.g. `markblocks -mark deletion -allow-partial -tenant "my-tenant" -backend gcs -gcs.bucket-name my-gcp-project 01GHF3BV8D6JNT8X88Q07F9FZK`. It cannot find the block as it is in fact looking at `gsutil ls -l gs://my-gcp-project/my-tenant/01GHF3BV8D6JNT8X88Q07F9FZK` i.e. not inside the `blocks` folder in the bucket
3. Run the following to actually delete the block correctly `markblocks -mark deletion -allow-partial -tenant "my-tenant/blocks" -backend gcs -gcs.bucket-name my-gcp-project 01GHF3BV8D6JNT8X88Q07F9FZK`. Note I have stuck the `blocks` path into the `tenant` flag to make it work.

#### Expected behavior

The "blocks" folder should be expected and the user should not have to append it to the tenant to force it to work

#### Environment

- Infrastructure: Kubernetes
- Deployment tool: Mimir Installed via Helm

#### Additional Context

Tested using a clone of `main` branch and also release `mimir-distributed-3.3.0`

**Comment by pstibrany** [comment]
Hello.

> 1. Mimir writes blocks to storage and we can retrieve them e.g. `gsutil ls -l gs://my-gcp-project/blocks/my-tenant/01GHF3BV8D6JNT8X88Q07F9FZK`

Do you use `-blocks-storage.storage-prefix` feature to write under "blocks" prefix? If so, you can also use `-storage-prefix` flag in markblocks and set it to `blocks`. Then using `-tenant=my-tenant` should work correctly.

**Comment by ronanbarrett** [comment]
Aha! I didn't realise this prefix flag had been set in the chart. Confirmed your suggested solution of using `-storage-prefix` works. Thanks.
